### DISCUSSION TITLE
Adding button dynamically

### DISCUSSION THREAD

**Discussion by Ondra09** [ArthurSonzogni/FTXUI#481] [Q&A]
Hi, 
I am reading some data over internet and I am doing it very slowly. When new information arrives, I would like to add new row in Container::Vertical. The row can look something like this:

| string | button |

String is static and I know how to add it. But If I add Button, it does not accepts input. If I try adding it do existing container by calling ComponentBase::Add() the renderer stops rendering what was already there and shows message: "Not implemented component"

Is it possible to add new buttons dynamically?

Thank you a lot.

**Comment by ArthurSonzogni** [comment]
> Is it possible to add new buttons dynamically?

Yes it is possible.
Could you paste some of your code. I believe I am missing informations here.

**Reply by Ondra09** [reply]
I took FromObject from json_tui project and create a line with button. This works and button accepts input. If I in render method add another lines to this container, it stops working. 

I was trying to update it from different thread as well, but result was the same, if I add something to buttons after it was constructed and main loop started it is not working. 


```
Component FromObject(Component comp, streaminfo::Stream &stream)
{
    class Impl : public ComponentBase
    {
    public:
        Impl(Component comp, WebInfo &info)
        {
            auto children = Container::Horizontal({});

            children->Add(Renderer([&] { return text(info.address); })|border);

            auto start = Button("Check", [](){});

            Add(Container::Horizontal({
                        comp, 
                        children,
                        start
                    }));
        }
    };

    return Make<Impl>(comp, stream);
}

auto lineContainer = Container::Horizontal({});
auto buttons = FromObject(lineContainer, list[0]);

auto headerRenderer = Renderer(buttons, [&]()
                                                 {
                                                     auto vlist = Container::Vertical({});
                                                     for (auto it = std::begin(list); it != std::end(list); it++)
                                                     {
                                                         if (it->status == streaminfo::Status::CHECKED)
                                                         {
                                                             auto nlineContainer = FromObject(lineContainer, *it);
                                                             vlist->Add(nlineContainer);
                                                         }
                                                     }

                                                     buttons->Add(vlist);

                                                     Elements elements({
                                                             separator(),
                                                             buttons->Render()
                                                         });

                                                     auto content = vbox(elements);

                                                     return window(text(APP_NAME), content);
                                                 });
auto screen = ScreenInteractive::Fullscreen();
screen.Loop(headerRenderer);
```

**Comment by pralad-p** [comment]
I've had a similar problem as well and the key is to pass your Component (in this a button) over to a Renderer as a child, so when you do include it in a vbox, it is trickled down in the component hierarchy and enables interactivity for the deeper nested components, and so on and so forth (depending at what depth, you still expect to observe interactivity).
